### PR TITLE
29 add limited standart collections support with some deviations from the original idea of the task

### PR DESCRIPTION
**Pull request by alexboost222**
**Subject:** Add List<T> support to ConfigsSerializationFacade + test refactoring

**What's done:**

Added support for List<T> where T can be any supported simple type
Fixed float/double parsing (was using NumberStyles.None, now uses NumberStyles.Float)
Refactored parser architecture using TryParse delegates and regex patterns
Reorganized tests into nested classes by type for better maintainability
Added comprehensive test coverage for edge cases and error scenarios

**Note on scope:**

Original task included type alias system (MyType -> System.Collections.Generic.List<int>) but that turned out overly complex for this iteration. Focused on core serialization improvements instead. Will handle namespace resolution with simple using System.Collections.Generic; in codegen as workaround.

### CODE REVIEW

**Review by vangogih — APPROVED** [review]
Good job, thanks!

There are some not critical comments from me.

Approve. You can resolve the comments and merge it 👍🏻

File: C4G.Unity/Assets/C4G/Tests/Editor/Unity/ConfigsSerializationFacadeTests.cs [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+using System.Linq;
 using C4G.Core;
 using C4G.Core.SheetsParsing;
 using C4G.Core.Utils;
 using NUnit.Framework;
 
 namespace C4G.Tests.Editor.Unity
 {
+    [TestFixture]
     public class ConfigsSerializationFacadeTests
```
[the remark this answers is not in the document]
Comment (alexboost222) [reply C4G.Unity/Assets/C4G/Tests/Editor/Unity/ConfigsSerializationFacadeTests.cs]: Sure, done!
